### PR TITLE
Add 'where did you hear about us?' dashboard

### PR DESCRIPTION
**Pull request by benlovell**
Adds a simple dashboard for querying these particular records. At the moment
this is by date range only, but could reasonably be extended to cover the
delivery partners or other criteria.

The duplication in the forms and CSV generators could do with some form of
abstraction soon, if we keep repeating them.

screen shot 2016-05-05 at 12 10 57

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
+  }
+
+  attributes.except!(:pension_provider) unless attributes[:where] == 'Pension Provider'
+
+  WhereDidYouHear.create!(attributes)
+  print '.'
+end
+
+puts
+puts 'done!'
```
Comment (dwhenry) [inline db/seeds.rb]: This feels more like a test fixture than a seed.

File: features/step_definitions/where_did_you_hear_about_us_report_steps.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+Given(/^there are existing where did you hear about us records$/) do
+  create_list(:where_did_you_hear, Kaminari.config.default_per_page + 1)
```
Comment (dwhenry) [inline features/step_definitions/where_did_you_hear_about_us_report_steps.rb]: Personal preference so happy to go withy the majority, but I prefer not to use the ```FactoryGirl``` magic and instead make it expicit.
```
FactoryGirl.create_list(...)
```

**Comment by dwhenry** [comment]
Does this report meet the user needs for the MFP report? 

The MFP report shows percentage breakdowns for the month for each WhereDidYouHear option, yet the report is a row for each data point in the list. 

Would we be better off doing some summarization?

File: db/seeds.rb (hunk 1 above)
Comment (benlovell) [inline db/seeds.rb]: How so? It creates 100 records of varying data from a closed set for the purposes of testing / demonstrating this feature during development / staging. Seems just like seeds to me.

File: features/step_definitions/where_did_you_hear_about_us_report_steps.rb (hunk 2 above)
Comment (benlovell) [inline features/step_definitions/where_did_you_hear_about_us_report_steps.rb]: My view on this is that everyone knows exactly what `create`, `build`, `create_list` mean in the context of a test so the prefix is noise. Why are these any different to other globally scoped things like `subject`, `let`, `before` etc? The global namespace in the context of tests is polluted to heck anyway.

**Comment by benlovell** [comment]
It doesn't attempt to cater for the specific needs of the MFP report, you're correct on that front.

I think we can deliver this as-is so we have a window into the Google Forms based 'where did you hear' data and there'll most certainly be some follow up work to add the month on month based stuff. What do you reckon?

**Comment by dwhenry** [comment]
🚢 

Can start to see some duplication - controllers/CSV generators - that we should extract in the near future.

**Pull request merged**